src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1763,6 +1710,234 @@ println outFile.text
 
 The complete description of user and global space APIs can be found in the http://doc.activeeon.com/javadoc/latest/org/ow2/proactive/scheduler/task/client/DataSpaceNodeClient.html[DataSpaceNodeClient JavaDoc].
 
+NOTE: The `userspaceapi` and `globalspaceapi` <<_glossary_script_bindings,script bindings>> can also be used in other <<_glossary_additional_task_scripts,scripts>> defined inside a <<_glossary_task,ProActive Task>>.
+To know in which scripts these bindings are available, refer to the <<_variables_quick_reference,variables quick reference>>.
+
+[[_task_synchronization_api]]
+=== Synchronization API
+
+Task synchronization is traditionally handled by the ProActive Scheduler through <<Dependency, Task Dependencies>>, where dependant ProActive Tasks are started only after their parents Tasks completed.
+
+In some cases, *static task dependencies* are not enough to synchronize tasks execution.
+
+This is the case when a task depends on the state of another task executed by a different branch of a workflow, a sibling task or even another job/workflow.
+
+For example, let's suppose workflow _A_ contains a single task _A_1_ starting an apache server. Task _A_1_ starts the server and remains alive while the server is up.
+When workflow _A_ is killed, task _A_1_ stops the apache server.
+
+A task _B_1_ of another workflow _B_ wants to interact with this server.
+
+*Problem:* task _B_1_ does not know if the apache server is up and how to contact it.
+
+Using *Task Synchonization API*, task _A_1_ can advertise the server URL to the Synchronization Service.
+
+Task _B_1_ can use Task Synchonization API to wait until the apache server is advertised, it can also perform the necessary operations and advertise task _A_1_ that the operation is complete and the apache server can be terminated.
+
+As we can see, Task Synchronization API can be used for two-ways synchronization.
+
+==== API Usage
+
+*Task Synchronization API* is a *Java API* defined by the following https://www.activeeon.com/public_content/documentation/javadoc/dev/org/ow2/proactive/scheduler/synchronization/Synchronization.html[API documentation].
+
+The Synchronization API is connected to a centralized *Synchronization Service*. This service ensures that each request is executed atomically, and that changes are persisted on disk.
+
+The *Synchronization Service* works as a _Key-Value Store_ organized in *channels*. Each *channel* acts as a private _Key-Value Store_ or _Hash Map_.
+
+This means that each _put_ operation is identified by a 3-tuple: _<Channel_identifier, Key_binding, Value>_.
+
+The Synchronization API is available inside ProActive Tasks through the `synchronizationapi` <<_glossary_script_bindings,script binding>>.
+
```

Comment:
We can start by saying that at startup, the synchronization service contains no channel.

Reply:
it contains only persistent channels previously created using the api